App Crash(After Install Not opening)

**Describe the bug**
A clear and concise description of what the bug is.
Can't open the app after installation
**To Reproduce**
Steps to reproduce the behavior:
1. Go to App 
2. It causes error and deny to open

**Expected behavior**
A clear and concise description of what you expected to happen.
Open and can be used by the user seamlessly.
**Screenshots**
If applicable, add screenshots to help explain your problem.

**Crash logs**
If possible, add crash logs to help us figure out the problem.

**Smartphone (please complete the following information):**
 - Device: [Samsung F34]
 - Andorid Version: [Android 14]
 - App Version: [v3.1.1]

**Additional context**
Add any other context about the problem here.

Same issue here, got a crash report extracted

```kotlin
FATAL EXCEPTION: main
Process: phone.vishnu.quotes, PID: 30875
java.lang.ExceptionInInitializerError
	at phone.vishnu.quotes.controller.AppController.attachBaseContext(SourceFile:20)
	at android.app.Application.attach(Application.java:361)
	at android.app.Instrumentation.newApplication(Instrumentation.java:1252)
	at android.app.LoadedApk.makeApplicationInner(LoadedApk.java:1503)
	at android.app.LoadedApk.makeApplicationInner(LoadedApk.java:1439)
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:7213)
	at android.app.ActivityThread.-$$Nest$mhandleBindApplication(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2292)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:240)
	at android.os.Looper.loop(Looper.java:351)
	at android.app.ActivityThread.main(ActivityThread.java:8423)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:584)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1013)
Caused by: java.lang.IllegalArgumentException: Q2.a is not an interface
	at java.lang.reflect.Proxy$ProxyClassFactory.apply(Proxy.java:635)
	at java.lang.reflect.Proxy$ProxyClassFactory.apply(Proxy.java:602)
	at java.lang.reflect.WeakCache$Factory.get(WeakCache.java:230)
	at java.lang.reflect.WeakCache.get(WeakCache.java:127)
	at java.lang.reflect.Proxy.getProxyClass0(Proxy.java:438)
	at java.lang.reflect.Proxy.newProxyInstance(Proxy.java:873)
	at F2.a.<clinit>(SourceFile:20)
	... 15 more
```

Same. Using Quotes 3.1.1 on OPPO R11s, with ColorOS v5.2.1 and Android 8.1.0.
With OPPO LogKit, I got a crash log. It seems that its main part is the same as the above one:

```
java.lang.ExceptionInInitializerError
	at phone.vishnu.quotes.controller.AppController.attachBaseContext(SourceFile:20)
	at android.app.Application.attach(Application.java:198)
	at android.app.Instrumentation.newApplication(Instrumentation.java:1104)
	at android.app.Instrumentation.newApplication(Instrumentation.java:1088)
	at android.app.LoadedApk.makeApplication(LoadedApk.java:1040)
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6125)
	at android.app.ActivityThread.-wrap1(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1839)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:197)
	at android.app.ActivityThread.main(ActivityThread.java:7022)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:515)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:837)
Caused by: java.lang.IllegalArgumentException: Q2.a is not an interface
	at java.lang.reflect.Proxy$ProxyClassFactory.apply(Proxy.java:589)
	at java.lang.reflect.Proxy$ProxyClassFactory.apply(Proxy.java:566)
	at java.lang.reflect.WeakCache$Factory.get(WeakCache.java:230)
	at java.lang.reflect.WeakCache.get(WeakCache.java:127)
	at java.lang.reflect.Proxy.getProxyClass0(Proxy.java:419)
	at java.lang.reflect.Proxy.newProxyInstance(Proxy.java:820)
	at F2.a.<clinit>(SourceFile:20)
	... 14 more
```

The complete log is 
[here](https://github.com/user-attachments/files/16479362/phone.vishnu.quotes-30807-2024-08-03-19-26-59.txt). However, I don't think the extra info is useful...

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -23,11 +23,12 @@ android {
             minifyEnabled true
             shrinkResources true
             proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
-        }
+			signingConfig signingConfigs.debug
+		}
     }
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_17
+        targetCompatibility JavaVersion.VERSION_17
     }
 	lintOptions {
 		checkReleaseBuilds false
@@ -73,12 +74,12 @@ dependencies {
     implementation "androidx.room:room-runtime:$room_version"
     annotationProcessor "androidx.room:room-compiler:$room_version"
 
-    implementation "ch.acra:acra-core:5.8.4"
+    implementation "ch.acra:acra-core:5.11.3"
 
 	implementation(platform("org.jetbrains.kotlin:kotlin-bom:1.8.0"))
 
-	compileOnly "com.google.auto.service:auto-service-annotations:1.0.1"
-    annotationProcessor "com.google.auto.service:auto-service:1.0.1"
+	compileOnly "com.google.auto.service:auto-service-annotations:1.1.1"
+    annotationProcessor "com.google.auto.service:auto-service:1.1.1"
 
     implementation 'androidx.constraintlayout:constraintlayout:2.1.4'
     implementation 'androidx.legacy:legacy-support-v4:1.0.0'

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -23,7 +23,6 @@ android {
             minifyEnabled true
             shrinkResources true
             proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
-			signingConfig signingConfigs.debug
 		}
     }
     compileOptions {

--- a/app/src/main/java/phone/vishnu/quotes/controller/AppController.java
+++ b/app/src/main/java/phone/vishnu/quotes/controller/AppController.java
@@ -48,6 +48,7 @@ public class AppController extends Application {
     @Override
     protected void attachBaseContext(Context base) {
         super.attachBaseContext(base);
+
         ACRA.init(
                 this,
                 new CoreConfigurationBuilder()

--- a/app/src/main/java/phone/vishnu/quotes/controller/AppController.java
+++ b/app/src/main/java/phone/vishnu/quotes/controller/AppController.java
@@ -48,10 +48,9 @@ public class AppController extends Application {
     @Override
     protected void attachBaseContext(Context base) {
         super.attachBaseContext(base);
-
         ACRA.init(
                 this,
-                new CoreConfigurationBuilder(this)
+                new CoreConfigurationBuilder()
                         .withBuildConfigClass(BuildConfig.class)
                         .withReportFormat(StringFormat.JSON));
     }

--- a/build.gradle
+++ b/build.gradle
@@ -2,7 +2,7 @@
 
 buildscript {
 
-    ext.kotlinVersion = '1.7.20'
+    ext.kotlinVersion = '1.9.0'
 
     repositories {
         google()

